Framework/Algorithms/src/UnwrapMonitor.cpp

```diff
@@ -109,23 +109,38 @@ void UnwrapMonitor::exec() {
       // If the detector flightpath is missing, zero the data
       g_log.debug() << "Detector information for workspace index " << i
                     << " is not available.\n";
-      tempWS->dataX(i).assign(tempWS->dataX(i).size(), 0.0);
-      tempWS->dataY(i).assign(tempWS->dataY(i).size(), 0.0);
-      tempWS->dataE(i).assign(tempWS->dataE(i).size(), 0.0);
+      tempWS->mutableX(i).assign(tempWS->x(i).size(), 0.0);
+      tempWS->mutableY(i).assign(tempWS->y(i).size(), 0.0);
+      tempWS->mutableE(i).assign(tempWS->e(i).size(), 0.0);
       continue;
     }
 
     // Unwrap the x data. Returns the bin ranges that end up being used
     const double Ld = L1 + spectrumInfo.l2(i);
-    const std::vector<int> rangeBounds = this->unwrapX(tempWS, i, Ld);
+    MantidVec newX;
+    const std::vector<int> rangeBounds = this->unwrapX(newX, i, Ld);
     // Unwrap the y & e data according to the ranges found above
-    this->unwrapYandE(tempWS, i, rangeBounds);
-    assert(tempWS->dataX(i).size() == tempWS->dataY(i).size() + 1);
+    MantidVec newY;
+    MantidVec newE;
+    this->unwrapYandE(tempWS, i, rangeBounds, newY, newE);
+
+    // Now set the new X, Y and E values on the histogram
+    auto histogram = tempWS->histogram(i);
+    histogram.resize(newY.size());
+    auto &resizedX = histogram.mutableX();
+    auto &resizedY = histogram.mutableY();
+    auto &resizedE = histogram.mutableE();
+    std::copy(newX.begin(), newX.end(), resizedX.begin());
+    std::copy(newY.begin(), newY.end(), resizedY.begin());
+    std::copy(newE.begin(), newE.end(), resizedE.begin());
+    tempWS->setHistogram(i, histogram);
+
+    assert(tempWS->x(i).size() == tempWS->y(i).size() + 1);
```

Comment:
According to the validators it will always be bin edges and counts:
```cpp
  wsValidator->add<HistogramValidator>();
  wsValidator->add<RawCountValidator>();
```

(unless this is changed at some point earlier in the algorithm run?)